### ISSUE TITLE
Bike improvement: support for smoothness tag

### ISSUE THREAD

**Issue by ratrun** [graphhopper/graphhopper#655]
The assumed highway speed should be adopted depending on http://wiki.openstreetmap.org/wiki/DE:Key:smoothness
The combination with all possible highway types, surface and grade tags needs to be considered.

**Comment by karussell** [comment]
This seems to be important but do you have an example where considering smoothness would improve the routing? Changing this without a real world example could make it worse.

**Comment by lukie80** [comment]
Smoothness is highly correlated with tracktype (grade#) but still allows for a refinement. The official definition (https://wiki.openstreetmap.org/wiki/Key:smoothness) is based on the physical usability of a way for wheeled vehicles:
- intermediate: good for regular bike
- bad: good for trekking bike / regular car
- very bad: only MTB and some effort  / high clearance car
- horrible: only MTB and high effort / Jeep

I would rate the following track as grade5 and very horrible if wet and grade4 and horrible if dry for example.
[Muddy forest track picture on pixabay.com](https://pixabay.com/en/forest-way-mud-track-tree-2771493/)

My personal statistical experience for German forest tracks in the flat country:
- grade1: good - intermediate, (default: good)
- grade2: good - intermediate, (default: intermediate)
- grade3: intermediate - bad, (no default)
- grade4: bad - very bad, (default: bad)
- grade5: bad (short grass on hard ground / lawn) - very horrible (overgrown, soft and uneven ground), (no default)

So basically, it mostly would alter the tracktype by +/-1 except for grade5 which could be lowered even more. One scoring possibility for bicycles could be:

- good and better: 24
- intermediate: 20
- bad: 15
- very bad: 6
- horrible and worse: 0
